solana/system_program.py

```diff
@@ -304,14 +335,50 @@ def sys_program_id() -> PublicKey:
     return PublicKey("11111111111111111111111111111111")
 
 
-def create_account(param: CreateAccountParams) -> Transaction:
+def create_account(params: CreateAccountParams) -> Transaction:
     """Generate a Transaction that creates a new account."""
-    raise NotImplementedError("create_account not implemented")
+    layout = SYSTEM_INSTRUCTION_LAYOUTS[CREATE_IDX]
+    data = encode_data(layout, params.lamports, params.space, params.program_id.__bytes__())
+
+    txn = Transaction()
+    txn.add(
+        TransactionInstruction(
+            keys=[
+                AccountMeta(pubkey=params.from_pubkey, is_signer=True, is_writable=True),
+                AccountMeta(pubkey=params.new_account_pubkey, is_signer=False, is_writable=True),
+            ],
+            program_id=sys_program_id(),
+            data=data,
+        )
+    )
+    return txn
 
 
 def assign(params: Union[AssignParams, AssignWithSeedParams]) -> Transaction:
     """Generate a Transaction that assigns an account to a program."""
-    raise NotImplementedError("assign not implemented")
+    if hasattr(params, "base_pubkey"):
+        params = cast(AssignWithSeedParams, params)
+        data = encode_data(
+            SYSTEM_INSTRUCTION_LAYOUTS[ASSIGN_WITH_SEED_IDX],
+            params.base_pubkey.__bytes__(),
```

Comment:
Instead of calling `__bytes__` directly, we can invoke it like this:

```python
bytes(params.base_pubkey)
```

Reply:
Fixed.